addons/l10n_in/models/account_invoice.py

```diff
@@ -71,6 +79,18 @@ def _compute_l10n_in_state_id(self):
             else:
                 move.l10n_in_state_id = False
 
+    @api.depends('partner_id', 'partner_shipping_id', 'company_id')
+    def _compute_fiscal_position_id(self):
+        indian_moves = self.env['account.move']
+        for move in self:
+            if move.country_code == 'IN' and move.journal_id.type == 'sale':
+                indian_moves += move
+                move.fiscal_position_id = self.env['account.fiscal.position'].with_company(move.company_id)._get_fiscal_position(
+                    move.partner_id,
+                    delivery=move._get_l10n_in_pos_sale_partner() or move.partner_id
+                )
```

Comment:
Less lines, less indentation, and better time complexity (`+=` on recordsets is O(n), so the loop was O(n²))
```suggestion
        indian_moves = self.filtered(lambda m: m.country_code == 'IN' and m.journal_id.type == 'sale')
        for move in indian_moves:
            move.fiscal_position_id = self.env['account.fiscal.position'].with_company(move.company_id)._get_fiscal_position(
                move.partner_id,
                delivery=move._get_l10n_in_pos_sale_partner() or move.partner_id
            )
```

Reply:
> Less lines, less indentation, and better time complexity (+= on recordsets is O(n), so the loop was O(n²))

@william-andre A bit curious to know isn't it the case filtered method browses the record in a loop? Than again we are traversing that filtered record set again to implement the logic. Shouldn't it would be better that one loop does the job with `+=` ?

Reply:
`O(n+n)` = `O(n)`. Looping twice doesn't change the general complexity (it won't diverge faster or slower depending on the size of the input)
On the other hand, recordsets are frozen. So every time you `+=`, you need to rebuild an entirely new one and reallocate memory. It makes it `O(n)` inside of a loop in `O(n)`, therefore `O(n*n)` = `O(n²)`